### PR TITLE
[LCE] End to End Model Conversion and Benchmarking Guide

### PR DESCRIPTION
**Pull request by jneeven**
~~Very much a draft, but I'm not entirely familiar with the doc syntax so I want to see how it looks. Feedback is welcome.~~

This is done from my side, let me know what you think

### CODE REVIEW

**Review by lgeiger — COMMENTED** [review]
Great! I think it would be cool to also add the "Run on Binder" and "View on GitHub" buttons to the top of the notebook so users can directly execute the notebook or view it on GitHub, similar to https://docs.larq.dev/larq/tutorials/mnist/.

**Comment by jneeven** [comment]
> Great! I think it would be cool to also add the "Run on Binder" and "View on GitHub" buttons to the top of the notebook so users can directly execute the notebook or view it on GitHub, similar to https://docs.larq.dev/larq/tutorials/mnist/.

Agreed, how do I obtain a binder URL for that?

**Comment by Tombana** [comment]
Should we also add a link in the main larq tutorial (not the compute-engine one)?
For example at the end of https://docs.larq.dev/larq/tutorials/mnist/ we could add "To benchmark larq models on ARM devices, please follow 'this guide'."

File: docs/compute-engine/end_to_end.ipynb [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+   "source": [
+    "This will produce the converted `quicknet.tflite` file with compressed weights and optimized operations, which is only just over 3 MiB in size!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Benchmarking\n",
+    "This part of the guide assumes that you'll want to benchmark on an Arm-64 based board such as a Raspberry Pi. For more detailed instructions on benchmarking, see the [Benchmarking guide](https://docs.larq.dev/compute-engine/benchmark).\n",
```
Comment (Tombana) [inline docs/compute-engine/end_to_end.ipynb:122]:
```suggestion
    "This part of the guide assumes that you'll want to benchmark on an Arm-64 based board such as a Raspberry Pi. For more detailed instructions on benchmarking and for benchmarking on Android phones, see the [Benchmarking guide](https://docs.larq.dev/compute-engine/benchmark).\n",
```

File: docs/compute-engine/end_to_end.ipynb [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    "    flatbuffer_bytes = lce.convert_keras_model(model)\n",
+    "    flatbuffer_file.write(flatbuffer_bytes)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that the model is converted, it is useful to visualize it in [netron](https://lutzroeder.github.io/netron/) to make sure the network looks as expected. For the model above, the first part of the flatbuffer looks like this:\n",
```
Comment (Tombana) [inline docs/compute-engine/end_to_end.ipynb:237]:
```suggestion
    "Now that the model is converted, it is useful to visualize it in [Netron](https://lutzroeder.github.io/netron/) to make sure the network looks as expected. Simply go to [Netron](https://lutzroeder.github.io/netron/) and select the `.tflite` file to visualize it. For the model above, the first part of the flatbuffer looks like this:\n",
```

**Review by Tombana — APPROVED** [review]
Great guide!

**Comment by jneeven** [comment]
> Should we also add a link to this tutorial in the main larq tutorial?
> For example at the end of https://docs.larq.dev/larq/tutorials/mnist/ we could add "To benchmark larq models on ARM devices, please follow 'this guide'."

Yeah that's a good idea, especially since it appears the existence / use of LCE is not entirely clear to some Larq users right now. I'll add it to this branch :+1: I'll also restructure the android guide today so that at the bottom of this guide I can link to the inference guides.

**Review by lgeiger — COMMENTED** [review]
Very nice guide, I especially like the netron screenshots.

Netron also has the option to display model graphs horizontally (`ctrl+K`). How do you feel about changing netron figures to show the graph horizontally to make the guide a bit more compact?

**Comment by jneeven** [comment]
> Very nice guide, I especially like the netron screenshots.
> 
> Netron also has the option to display model graphs horizontally (`ctrl+K`). How do you feel about changing netron figures to show the graph horizontally to make the guide a bit more compact?

I already wanted to do this, but didn't know it was possible :exploding_head: Will do this later today

**Review by lgeiger — APPROVED** [review]

**Pull request merged**